packages/client/src/hooks/useForm.tsx

```diff
@@ -0,0 +1,86 @@
+import React, { useCallback, useState } from 'react'
+import { Field } from '../components/UI/Field'
+import { FieldType } from '../types/field'
+
+export const useForm = (config: FieldType[]): any[] => {
+  const initialData: Record<string, any> = {}
+  const initialValues: Record<string, any> = {}
+  for (const field of config) {
+    initialData[field.name] = field
+    initialValues[field.name] = field.value
+  }
+  const [formData, setFormData] = useState(initialData)
+
+  const handleChange = (e: React.FormEvent<HTMLInputElement>) => {
+    const name = e.currentTarget.name
+    const value = e.currentTarget.value
+    const fieldData = { ...formData[name] }
+
+    if (!fieldData.isValid) {
+      fieldData.isValid = fieldData.rule.test(fieldData.value)
+    }
+
+    fieldData.value = value
+
+    setFormData({
+      ...formData,
+      [name]: fieldData
+    })
+  }
+
+  const formValues = useCallback(() => {
+    const values: Record<string, any> = {}
+    const fields = Object.keys(formData)
+
+    for (const i of fields) {
+      values[i] = formData[i].value
+    }
+
+    return values

```

Comment:
Почему бы не сделать просто так:

```suggestion
    const values = Object.values(formData)
```
?
Если формдата, это объект FormData, то ещё проще [formData.values()](https://developer.mozilla.org/en-US/docs/Web/API/FormData/values);

Reply:
в таком случае я получу не то, что мне нужно, мне надо достать поле value, а остальные поля не нужны для отправки на сервер

Reply:
тут тоже что Object.keys убрала

Reply:
> в таком случае я получу не то, что мне нужно, мне надо достать поле value, а остальные поля не нужны для отправки на сервер

```
Object.values(formData).map(({ value }) => value);
```

Reply:
не понимаю, так же получится массив, а мне нужен объект

Reply:
хм. сори, чуть поспешил. В таком случае можно писать через reduce:

```js
Object
  .entries(formData)
  .reduce((acc, [field, fieldValue]) => {
    acc[field] = fieldValue.value;
    return acc;
  }, {});
```

Понимаю что код выглядит чуть сложнее на первый взгляд, но это всё синтаксические сахары JS-а, нужно ими пользоваться, [reduce](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/reduce) очень крутая штука;

Reply:
спасибо! сделала так